lms/templates/courseware/course_navigation.html

```diff
@@ -65,102 +67,46 @@
-        selectElement.attr("title", "${_("Course is not yet visible to students.")}");
-    % endif
-
-    % if specific_student_selected:
-        userNameContainer.css('display', 'inline-block');
-        userNameElement.val('${masquerade_user_name}');
-    % endif
-
-    selectElement.change(function() {
-        var selectedOption;
-        if (selectElement.attr("disabled")) {
-           return alert("${_("You cannot view the course as a student or beta tester before the course release date.")}");
-        }
-        selectedOption = selectElement.find('option:selected');
-        if (selectedOption.val() === 'specific student') {
-            userNameContainer.css('display', 'inline-block');
-        } else {
-            userNameContainer.hide();
-            masquerade(selectedOption);
-        }
-    });
-
-    userNameElement.keypress(function(event) {
-        if (event.keyCode === 13) {
-            // Avoid submitting the form on enter, since the submit action isn't implemented.  Instead, blur the
-            // element to trigger a change event in case the value was edited, which in turn will trigger an AJAX
-            // request to update the masquerading data.
-            userNameElement.blur();
-            return false;
-        }
-        return true;
-    });
-
-    userNameElement.change(function() {
-        masquerade(selectElement.find('option:selected'));
-    });
-
-    function masquerade(selectedOption) {
-        var data = {
-            role: selectedOption.val() === 'staff' ? 'staff' : 'student',
-            user_partition_id: ${cohorted_user_partition.id if cohorted_user_partition else 'null'},
-            group_id: selectedOption.data('group-id'),
-            user_name: selectedOption.val() === 'specific student' ? userNameElement.val() : null
-        };
-        $.ajax({
-            url: '/courses/${course.id}/masquerade',
-            type: 'POST',
-            dataType: 'json',
-            contentType: 'application/json',
-            data: JSON.stringify(data),
-            success: function(result) {
-                if (result.success) {
-                    location.reload();
-                } else {
-                    alert(result.error);
-                }
-            },
-            error: function() {
-                alert('Error: cannot connect to server');
-            }
-        });
+    <%
+    preview_options = {
+        "courseId": course.id,
+        "disableStudentAccess": disable_student_access if disable_student_access is not UNDEFINED else False,
+        "specificStudentSelected": specific_student_selected,
+        "cohortedUserPartitionId": cohorted_user_partition.id if cohorted_user_partition else None,
+        "masqueradeUsername" : masquerade_user_name if masquerade_user_name is not UNDEFINED else None,
     }
-}());
-</script>
+    %>
+    <%static:require_module module_name="lms/js/preview/preview_factory" class_name="PreviewFactory">
```

Comment:
Reviewers: I considered bundling this logic into a pre-existing bundle rather than adding another download to the page. However, that made the logic more complicated, and this file is only loaded for staff users.